File: leet_code/array/136/136.dart
class Solution {
  int singleNumber(List<int> nums) {
    final Set<int> seen = {};
    for (int num in nums) {
      if (seen.contains(num)) {
        seen.remove(num);
      } else {
        seen.add(num);
      }
    }
    return seen.first;
  }
}

void main() {
  Solution s = Solution();
  print(s.singleNumber([4, 2, 2, 1, 1]));
}

File: leet_code/array/136/136.py
from typing import List
class Solution:
    def singleNumber(self,nums:List[int])->int:
        seen = set()
        for num in nums:
            if num in seen:
                seen.remove(num)
            else:
                seen.add(num)
        return seen.pop()

s =Solution()
print(s.singleNumber([4,2,1,2,1]))